tests/conftest.py

```diff
@@ -0,0 +1,76 @@
+"""Common fixtures"""
+import pytest
+
+from alpenhorn import config, db, extensions
+
+
+def pytest_configure(config):
+    config.addinivalue_line(
+        "markers",
+        "alpenhorn_config(*config_dict): "
+        "used to set the alpenhorn.config for testing.  config_dict"
+        "is merged with the default config.",
+    )
+
+
+@pytest.fixture
+def set_config(request):
+    """Set alpenhorn.config.config for testing.
+
+    Any value given in the alpenhorn_config mark is merged into the
+    default config.
+
+    Yields alpenhorn.config.config.
+
+    After the test completes, alpenhorn.config.config is set to None."""
+    # If initialise with the default
+    config.config = config._default_config.copy()
+
+    marker = request.node.get_closest_marker("alpenhorn_config")
+    if marker is not None:
+        config.config = config.merge_dict_tree(config.config, marker.args[0])
+
+    yield config.config
+
+    # Reset globals
+    config.config = None
+    extensions._db_ext = None
+    extensions._ext = None
+
+
+@pytest.fixture
+def use_chimedb(set_config):
+    """Use chimedb, if possible.
+
```

Comment:
It might be worth pointing out that this doesn't actually run against the live chimedbrc.